Drivers/NER/src/can.c

```diff
@@ -325,11 +340,22 @@ void HAL_CAN_RxFifo1MsgPendingCallback(CAN_HandleTypeDef *hcan)
 {
     CAN_RxHeaderTypeDef* rx_header = malloc(sizeof(CAN_RxHeaderTypeDef));
     can_msg_t new_msg;
-    new_msg.line = CAN_LINE_2;
-    HAL_CAN_GetRxMessage(hcan, CAN_RX_FIFO1, rx_header, new_msg.data);
+    new_msg.line = CAN_LINE_1;
+    HAL_CAN_GetRxMessage(hcan, CAN_RX_FIFO0, rx_header, new_msg.data);
     new_msg.len = rx_header->DLC;
     new_msg.id = rx_header->StdId;
-    enqueue(can2_incoming, new_msg);
+    switch(new_msg.id)
+    {
+        case 0x2010:  
```

Comment:
I made this change to remove all the function calls, you can cascade switch statements like this